internal/build/imgsrc/depot.go

```diff
@@ -208,15 +209,27 @@ func initBuilder(ctx context.Context, buildState *build, appName string, streams
 
 	// Set the buildErr to any error that represents the build failing.
 	var buildErr error
+	var finalBuildErr error
 
 	span.AddEvent("Acquiring Depot machine")
 	var buildkit *depotmachine.Machine
-	buildkit, buildErr = depotmachine.Acquire(ctx, build.ID, build.Token, "amd64")
-	if buildErr != nil {
-		build.Finish(buildErr)
+
+	timeoutCtx, cancel := context.WithTimeout(ctx, 25*time.Second)
+	defer cancel()
+
+	finalBuildErr = retry.Retry(timeoutCtx, func() error {
+		buildkit, buildErr = depotmachine.Acquire(ctx, build.ID, build.Token, "amd64")
+		if buildErr != nil {
+			span.RecordError(buildErr)
```

Comment:
Do you want this to live outside the retry? I'm thinking that this will mark the span as errored even if the error is temporary

Reply:
This will add an exception message in the span, so we can still track errors even if the build ends up succeeding after retries. If we can discover that a build succeeded, that would help differentiate.